### PR TITLE
remove explain button and refactor a bit

### PR DESCRIPTION
**Pull request by imbananko**

### CODE REVIEW

**Comment by imbananko** [comment]
#53

File: src/main/kotlin/com/imbananko/tilly/MemeManager.kt [hunk 1]
```diff
@@ -73,12 +71,12 @@ class MemeManager(private val memeRepository: MemeRepository, private val voteRe
...
     runCatching {
-      val memeOfTheWeek: MemeEntity? = memeRepository.findMemeOfTheWeek(chatId)
+      val memeOfTheWeek: MemeEntity = memeRepository.findMemeOfTheWeek(chatId)!!
 
-      if (memeOfTheWeek != null) {
+      run {
         val winner = execute(GetChatMember().apply {
           this.setChatId(memeOfTheWeek.chatId)
-          this.setUserId(memeOfTheWeek.senderId)
+          this.userId = memeOfTheWeek.senderId
```
Comment (yeplastun) [inline src/main/kotlin/com/imbananko/tilly/MemeManager.kt]: `this` is not necessary

Comment (yeplastun) [reply src/main/kotlin/com/imbananko/tilly/MemeManager.kt]: setChatId can be replaced in the same way as userId

File: src/main/kotlin/com/imbananko/tilly/MemeManager.kt [hunk 2]
```diff
@@ -146,13 +137,18 @@ class MemeManager(private val memeRepository: MemeRepository, private val voteRe
...
       }.onSuccess { sentMemeMessage ->
         val meme = MemeEntity(sentMemeMessage.chatId, sentMemeMessage.messageId, message.from.id, message.photo[0].fileId)
-        log.info("Sent meme=$meme")
         memeRepository.save(meme)
+
+        log.info("Sent meme=$meme")
       }.onFailure { throwable ->
-        log.error("Failed to send meme from message=$message. Exception=", throwable)
+        if (throwable is TelegramApiRequestException) {
+          log.error("Failed to send meme from message=$message. Exception=${throwable.apiResponse}")
```
Comment (yeplastun) [inline src/main/kotlin/com/imbananko/tilly/MemeManager.kt]: `logger.error("bla bla bla", throwable)` should do the thing

**Pull request closed without merging**